Bottom: 3425768aeef91280ffe856fcdf1ebab4c0c24952
Top:    6ad31392821d7654d152ed9bd430ad3232353ab8
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-24 01:10:02 +0100

draw.c: Draw pins, pads and vias (for non-gui HIDs) from DrawLayerGroup()


---

diff --git a/src/draw.c b/src/draw.c
index d096377..2ebab3e 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -83,7 +83,7 @@ static const BoxType *clip_box = NULL;
 static void Redraw (bool, BoxTypePtr);
 static void DrawEverything (BoxTypePtr);
 static void DrawPPV (int group, const BoxType *);
-static int DrawLayerGroup (int, const BoxType *);
+static void DrawLayerGroup (int, const BoxType *);
 static void DrawPinOrViaLowLevel (PinTypePtr, bool);
 static void DrawPlainPin (PinTypePtr, bool);
 static void DrawPlainVia (PinTypePtr, bool);
@@ -315,7 +315,6 @@ PrintAssembly (int side, const BoxType * drawn_area)
 
   gui->set_draw_faded (Output.fgGC, 1);
   DrawLayerGroup (side_group, drawn_area);
-  DrawPPV (side_group, drawn_area);
   gui->set_draw_faded (Output.fgGC, 0);
 
   /* draw package */
@@ -376,8 +375,7 @@ DrawEverything (BoxTypePtr drawn_area)
       int group = drawn_groups[i];
 
       if (gui->set_layer (0, group, 0))
-        if (DrawLayerGroup (group, drawn_area) && !gui->gui)
-          DrawPPV (group, drawn_area);
+        DrawLayerGroup (group, drawn_area);
     }
 
   if (TEST_FLAG (CHECKPLANESFLAG, PCB) && gui->gui)
@@ -813,7 +811,7 @@ DrawLayer (LayerTypePtr Layer, const BoxType * screen)
  * draws one layer group.  Returns non-zero if pins and pads should be
  * drawn with this group.
  */
-static int
+static void
 DrawLayerGroup (int group, const BoxType *drawn_area)
 {
   int i, rv = 1;
@@ -823,6 +821,7 @@ DrawLayerGroup (int group, const BoxType *drawn_area)
   Cardinal *layers = PCB->LayerGroups.Entries[group];
 
   clip_box = drawn_area;
+
   for (i = n_entries - 1; i >= 0; i--)
     {
       layernum = layers[i];
@@ -835,7 +834,9 @@ DrawLayerGroup (int group, const BoxType *drawn_area)
     }
   if (n_entries > 1)
     rv = 1;
-  return rv;
+
+  if (rv == 1 && !gui->gui)
+    DrawPPV (group, drawn_area);
 }
 
 /* ---------------------------------------------------------------------------
